cool.py

```diff
@@ -0,0 +1,294 @@
+
+        return np.array(None)
+
+    #: Set up parameters for images, mosaic, number of cols/rows (every image will be 256x256)
+    tile_width = 256
+    number_columns = 2
+    number_rows = 2
+
+    #: Build mosaic image with white background
+    total_height = tile_width * number_rows
+    total_width = tile_width * number_columns
+
+    logging.info("mosaicking images for %s", tile_name)
+
+    mosaic_image = np.zeros((total_height, total_width, 3), dtype=np.uint8)
+    mosaic_image[:, :] = (255, 255, 255)
+
+    i = 0
+    for tile in tiles:
+        #: convert from bytes to cv2
+        img = convert_to_cv2_image(tile)
+        #: Add image into the mosaic
+        row_start = (math.floor(i / number_columns)) * tile_width
+        col_start = (i % number_columns) * tile_width
+        mosaic_image[row_start : row_start + tile_width, col_start : col_start + tile_width] = img
+
+        i += 1
+
+    if out_dir:
+        if not out_dir.exists():
+            out_dir.mkdir(parents=True)
+
+        mosaic_outfile = out_dir / f"{tile_name}.jpg"
+        logging.debug("saving to %s", mosaic_outfile)
+        cv2.imwrite(str(mosaic_outfile), mosaic_image)
+
+        return mosaic_image
+
+    else:
+        logging.debug("no output directory provided")
+
+        return mosaic_image
+
+
+def load_pytorch_model():
+    """load pytorch model with tower scout weights
+
+    Args:
+        None
+
+    Returns:
+        pytorch model: model ready for scanning
+    """
+    model_weight_path = Path(__file__).parent / 'tower_scout' / 'xl_250_best.pt'
+    yolov5_path = Path(__file__).parent / 'yolov5'
+    model = torch.hub.load(str(yolov5_path), 'custom', path=str(model_weight_path), source='local')
+    #: if statements (check that .pt file exists, folders, etc.)
+    
+    return model
+
+
+def get_model():
+    """gets pytorch model using logic to ensure it's only loaded once
+
+    Args:
+        None
+
+    Returns:
+        model: loaded pytorch model ready for use
+    """
+    global MODEL
+    if MODEL is None:
+        logging.info('loading pytorch model')
+        MODEL = load_pytorch_model()
+
+    return MODEL
+
+
+def detect_towers(image):
+    """run pytorch model with tower scout weight on an image to detect cooling towers
+
+    Args:
+        image (obj): Path object to local image file or np.ndarray object of in-memory file
+
+    Returns:
+        result (obj): pytorch result object
+    """
+    towerscout_model = get_model()
+        
+    # if isinstance(image, Path):
+    #     logging.info('working with a Path')
+    #     pil_image = Image.open(image)
+    # elif isinstance(image, np.ndarray):
+    #     logging.info('working with a numpy array')
+    #     pil_image = Image.fromarray(image)
+
+    logging.info('working with %s', type(image))
+
+    scan_start = perf_counter()
+    results = towerscout_model(image)
+    logging.info('time to scan image: %s',  format_time(perf_counter() - scan_start))
+
+    results.save()
```

Comment:
**issue**: this should be removed or put behind a cli flag somehow

Reply:
moved to CLI